src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -589,12 +589,13 @@ private void executeSql(final Connection connection, final String sql, final boo
       }
    }
 
-   private void createNetworkTimeoutExecutor(final DataSource dataSource, final String dsClassName, final String jdbcUrl)
+   private void createNetworkTimeoutExecutor(final DataSource dataSource, final String dsClassName, final String jdbcUrl, final String driverClassName)
    {
       // Temporary hack for MySQL issue: http://bugs.mysql.com/bug.php?id=75615
       if ((dsClassName != null && dsClassName.contains("Mysql")) ||
           (jdbcUrl != null && jdbcUrl.contains("mysql")) ||
-          (dataSource != null && dataSource.getClass().getName().contains("Mysql"))) {
+          (dataSource != null && dataSource.getClass().getName().contains("Mysql")) ||
+          "com.amazonaws.secretsmanager.sql.AWSSecretsManagerMySQLDriver".equals(driverClassName)) {
```

Comment:
In your case, what is the dataSource classname? Does it contain the name "MySQL", if so the existing contains("Mysql") maybe should just be changed to a case insensitive check.